### PR TITLE
Document implicit pipelining

### PR DESCRIPTION
**Pull request by nussjustin**
Updates #131

### CODE REVIEW

File: radix.go [hunk 1]
```diff
@@ -142,6 +142,29 @@
...
+//
+// Currently only pools returned by NewPool implement implicit pipelining.
+// This includes the pools used by default in NewCluster and NewSentinel,
+// thus both Cluster and Sentinel can profit from implicit pipelining.
+//
+// See the documentation on Pool for more information about the current
+// implementation of implicit pipelining and for how to configure or disable
+// the feature.
+//
+// For a performance comparisons between Clients with and without impliciz
```
Comment (cep21) [inline radix.go]: Spelling `impliciz`

File: radix.go [hunk 2]
```diff
@@ -142,6 +142,29 @@
 //
 // Use the golang.org/x/xerrors package if you're using an older version of go.
 //
+// Implicit pipelining
+//
+// Implicit pipelining is an optimization that reduces the overhead of
```
Comment (mediocregopher) [inline radix.go]: I try to keep the main package doc very high level, and stick to only the bare minimum info that someone needs in order to use radix, or to preemptively answer questions that we get a lot (e.g. about AUTH). So I think these three paragraphs are a bit too wordy, but could be rolled together into one, like this:

> Implicit pipelining is an optimization implemented and enabled in the default Pool implementation (and therefore also used by Cluster and Sentinel) which involves delaying concurrent Cmds and FlatCmds a small amount of time and sending them to redis in a single batch, similar to manually using a Pipeline. By doing this radix significantly reduces the I/O and CPU overhead for concurrent requests.

> Note that only commands which do not block are eligible for implicit pipelining.

> See the documentation on Pool etc etc

What do you think?

**Comment by mediocregopher** [comment]
@cep21 did you have a chance to check out the changes I made to the `CmdAction` docs? I tried to make it clearer how `CmdAction` interacts with pipelining (implicit or otherwise), hopefully I did a better job than before.

File: radix.go (hunk 1 above)
Comment (nussjustin) [reply radix.go]: Done

File: radix.go (hunk 2 above)
Comment (nussjustin) [reply radix.go]: Sounds good to me. I updated the branch.

**Comment by cep21** [comment]
@mediocregopher I'm looking it over.  Give me a few days and I'll get back to you with my thoughts.

**Comment by cep21** [comment]
Oops dropped the ball here :) Thanks for the comments!
